src/devices/MotorHat/MotorHat.cs

```diff
@@ -0,0 +1,180 @@
+
+namespace Iot.Device.MotorHat
+{
+    /// <summary>
+    /// Raspberry Pi Motor Hat based on PCA9685 PWM controller
+    /// </summary>
+    public class MotorHat : IDisposable
+    {
+        /// <summary>
+        /// Default I2C address of Motor Hat
+        /// </summary>
+        public const int I2cAddressBase = 0x60;
+
+        /// <summary>
+        /// Motor Hat is built on top of a PCa9685
+        /// </summary>
+        private Pca9685 _pca9685;
+
+        /// <summary>
+        /// Holds every channel that is being used by either a DCMotor, Stepper, ServoMotor, or PWM
+        /// </summary>
+        private List<PwmChannel> channelsUsed = new List<PwmChannel>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MotorHat"/> class with the specified I2C settings and PWM frequency.
+        /// </summary>
+        /// <param name="settings">The I2C settings of the MotorHat.</param>
+        /// <param name="frequency">The frequency in Hz to set the PWM controller.</param>
+        /// <remarks>
+        /// The default i2c address is 0x60, but the HAT can be configured in hardware to any address from 0x60 to 0x7f.
+        /// The PWM hardware used by this HAT is a PCA9685. It has a total possible frequency range of 24 to 1526 Hz.
+        /// Setting the frequency above or below this range will cause PWM hardware to be set at its maximum or minimum setting.
+        /// </remarks>
+        public MotorHat(I2cConnectionSettings settings, double frequency)
+        {
+            var device = I2cDevice.Create(settings);
+            this._pca9685 = new Pca9685(device);
+
+            _pca9685.PwmFrequency = frequency;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MotorHat"/> class with the default I2C address and PWM frequency.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="MotorHat"/> will be created with the default I2C address of 0x60 and PWM frequency of 1600 Hz.
+        /// </remarks>
+        public MotorHat(double frequency = 1600) : this(new I2cConnectionSettings(1, I2cAddressBase + 0b000000), frequency)
+        {
+            // Use the following code to set an address equivalent to the one configured in your device jumpers
+            // var busId = 1;
+            // var selectedI2cAddress = 0b000000;     // A5 A4 A3 A2 A1 A0
+            // var deviceAddress = Pca9685.I2cAddressBase + selectedI2cAddress;
+            // var settings = new I2cConnectionSettings(busId, deviceAddress);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="DCMotor"/> object for the specified channel.
+        /// </summary>
+        /// <param name="motorNumber">A motor number from 1 to 4.</param>
+        /// <remarks>
+        /// The motorNumber parameter refers to the motor numbers M1, M2, M3 or M4 printed in the device.
+        /// </remarks>
+        public DCMotor.DCMotor CreateDCMotor(int motorNumber)
```

Comment:
unless 1..4 is mentioned in the MotorHat specs or somewhere please start numbering with 0